libraries/standard/http/src/http_client.c

```diff
@@ -155,21 +155,14 @@ static HTTPStatus_t _addHeader( HTTPRequestHeaders_t * pRequestHeaders,
                                  ( int32_t ) fieldLen, pField,
                                  ( int32_t ) valueLen, pValue );
 
-        if( ( bytesWritten + HTTP_HEADER_LINE_SEPARATOR_LEN ) != toAddLen )
-        {
-            IotLogErrorWithArgs( "Internal error in snprintf() in _addHeader(). "
-                                 "BytesWritten: %d.", bytesWritten );
-        }
-        else
-        {
-            pBufferCur += bytesWritten;
+        assert( bytesWritten == (int) ( toAddLen - HTTP_HEADER_LINE_SEPARATOR_LEN ) );
```

Comment:
We are not asserting here before the input into snprintf is different than the integers we were writing in AddRangeHeader().
The inputs into snprintf here are from user inputs, so we need to bubble up the encoding error.